src/bot/commands/report.ts

```diff
@@ -0,0 +1,409 @@
+import { Category, CategoriesRefsFull, ReportReasons, UnsignedJoke, Joke, Reason } from '../../typings';
+import prisma from '../../prisma';
+import { ProposalType } from '@prisma/client';
+import { Colors, commandsChannelId } from '../constants';
+import Command from '../lib/command';
+import { compareTwoStrings } from 'string-similarity';
+import { interactionInfo, interactionProblem, info, messageProblem, tDelete, isEmbedable } from '../utils';
+import { reportsChannelId } from '../constants';
+
+enum IdType {
+  MESSAGE_ID,
+  JOKE_ID,
+  MESSAGE_QUESTION
+}
+
+interface JokeCorrectionPayload extends UnsignedJoke {
+  id?: number;
+  correction_type: ProposalType;
+  suggestion: UnsignedJoke & {
+    message_id: string | null;
+    proposal_id: number;
+  };
+}
+
+export default class ReportCommand extends Command {
+  constructor() {
+    super({
+      name: 'report',
+      description: 'Signaler une blague',
+      type: ApplicationCommandType.ChatInput,
+      options: [
+        {
+          type: ApplicationCommandOptionType.Integer,
+          name: 'id',
+          description: 'ID de la blague',
+          required: true
+        },
+        {
+          type: ApplicationCommandOptionType.String,
+          name: 'raison',
+          description: 'Raison du signalement de la blague',
+          required: true,
+          choices: Object.entries(ReportReasons).map(([key, name]) => ({
+            name,
+            value: key
+          }))
+        },
+        {
+          type: ApplicationCommandOptionType.String,
+          name: 'informations',
+          description: 'Informations supplémentaires',
+          required: false
+        }
+      ]
+    });
+  }
+  async run(interaction: ChatInputCommandInteraction<'cached'>) {
+    if (interaction.channelId !== commandsChannelId) {
+      return interaction.reply(
+        interactionInfo(`Préférez utiliser les commandes dans le salon <#${commandsChannelId}>.`)
+      );
+    }
+
+    // TODO : ajouter un choix pour signaler une blague refusée injustement + dans approve, mettre le lien de la blague dans le "Votre approbation a bien été retirée."
+
+    const jokeId = interaction.options.getInteger('id', true);
+    const joke = jokeById(jokeId);
+    if (!joke) {
+      return interaction.reply(interactionInfo(`L'identifiant \`${jokeId}\` ne correspond à aucune blague connue.`));
+    }
+
+    const raison = interaction.options.getString('raison', true);
+
+    const embed = {
+      author: {
+        name: interaction.user.tag,
+        icon_url: interaction.user.displayAvatarURL({
+          size: 32
+        })
+      },
+      fields: [
+        {
+          name: 'Blague signalée',
+          value: `
+          > **Type**: ${CategoriesRefsFull[joke.type]}
+          > **Blague**: ${joke.joke}
+          > **Réponse**: ${joke.answer}
+          `,
+          inline: true
+        }
+      ],
+      color: Colors.PROPOSED
+    };
+
+    if (raison === 'doublon') {
```

Comment:
Faire un switch avec un return vers une fonction `this.reportDuplicate(...)`

Reply:
J'ai pas compris où tu veux en venir... Tu pourrais m'expliquer stp ?

Reply:
```ts
switch(reason) {
  case ''dupplicate': return this.reportDuplicate(...)
  case ''inappropriate': return this.reportInappropriate(...)
}
```